src/pymovements/dataset/dataset_files.py

```diff
@@ -318,16 +272,75 @@ def load_gaze_file(
     """
     if custom_read_kwargs is None:
         custom_read_kwargs = {}
+    add_columns = {
+        key: fileinfo_row[key] for key in
+        [key for key in fileinfo_row.keys() if key != 'filepath']
+    }
 
     if filepath.suffix in {'.csv', '.txt', '.tsv'}:
         if preprocessed:
-            gaze_df = pl.read_csv(filepath)
+            gaze_df = from_csv(
+                filepath,
+                trial_columns=definition.trial_columns,
+                add_columns=add_columns,
+                column_dtypes=definition.filename_format_dtypes,
+            )
+
+            pixel_columns: list[str] = list(
+                set(GazeDataFrame.valid_pixel_position_columns) & set(gaze_df.frame.columns),
```

Comment:
let's drop the usage of  `GazeDataFrame.valid_pixel_position_columns` here, as the preprocessed csv files are unnested using `GazeDataFrame.unnest()` before, so we actually know all the suffixes that can appear.

I now also see that the previous code with computing the union of two sets of column names is a bit error-prone, as we currently have to take care of the right order when passing `pixel_columns`.

Instead, let's write it this way:

```python
# suffixes as ordered after using GazeDataFrame.unnest()
component_suffixes = ['x', 'y', 'xl', 'yl', 'xr', 'yr', 'xa', 'ya']

pixel_columns = ['pixel_' + suffix for suffix in component_suffixes]
pixel_columns = [c for c in pixel_columns if c in gaze_df.frame.columns]
```

The same can be done for `position_columns`, `velocity_columns` and `acceleration_columns`

Reply:
done